### DISCUSSION TITLE
How to specify Platform in DotNetBuild with Cake 3

### DISCUSSION THREAD

**Discussion by jpiechowiak** [cake-build/cake#4205] [Q&A] [answered]
Hi,

Im converting `Xamarin` app to `Maui` and for that reason I've switched to Cake 3 and `DotNetBuild` command.
In my iOS app csproj I've `PropertyGroup` defined like that:

```xml
<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|iPhoneSimulator' ">
```

now I'm trying to build with following code:

```cake
var msBuildSettings = new DotNetMSBuildSettings()
  .WithTarget("Build")
  .WithProperty("ArchiveOnBuild", "True")
  .WithProperty("IpaPackageDir", ipaPackageDir)
  .WithProperty("Platform", platform);

DotNetBuild("mycsproj.csproj", new DotNetBuildSettings{
  NoRestore = true,
  Configuration = configuration,
  MSBuildSettings = msBuildSettings
});
```

But It looks that `Platform` is not recognised correctly. 
According to this issue https://github.com/dotnet/sdk/issues/9966

something like this suppose to be working:
`dotnet build /p:Platform=iPhoneSimulator`

Any ideas how to achieve this in Cake?

**Comment by nils-a** [comment]
@jpiechowiak can you elaborate on what "recognize" means in your question? 

The example you gave will produce something akin to the following `"C:/Program Files/dotnet/dotnet.exe" build "mycsproj.csproj" --configuration Release --no-restore /target:Build /property:ArchiveOnBuild=True /property:IpaPackageDir=[ipaPackageDir] /property:Platform=[platform]` 

The code you provided will set a property named `Platform` to the given value during build. There is no "recognizing" involved or needed. 

I would suggest you try to build your app "manually", using `dotnet build`. If you have a working commandline, people here can certainly help you to create a Cake script that generates the required commandline.

**Reply by jpiechowiak** [reply]
Ok. In general I want to match specific PropertyGroup from sample based on value provided for configuration and platform (maybe 'recognise' wasn't a perfect word:)

So it suppose to be
Configuration=Release
Platform=iPhoneSimulator

In my csproj I've two `PropertyGroup`.
First `Release|iPhone `
Second `Release|iPhoneSimulator`.

In my scenario first group is matched (based only on configuration). For provided data it should be second one.
But based on output you provided, parameters seems to look correct.

I will try again tomorrow.

**Reply by nils-a** [reply]
Hint: You can show the commands "created" by cake by adding the [`--settings_showprocesscommandline=true` argument](https://cakebuild.net/docs/running-builds/configuration/default-configuration-values#show-process-command-line) when running cake.

**Comment by jpiechowiak** [comment, marked as answer]
After further investigation, it looks like that it is probably not related to Cake, but some botnet issues. 
@nils-a  thanks for hint with argument. It helped me to know what is going on.
